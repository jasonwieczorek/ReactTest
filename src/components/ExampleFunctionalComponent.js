import React from "react";

/**
 * An example Functional Component. This is way more concise compared to the equivelant Class Component!
 */
export default function ExampleFunctionalComponent() {

    // Functional components use React hooks to handle state
    const[inputText, setInputText] = React.useState('');

    // equivalent of componentDidMount
    React.useEffect(() => {
       console.log('functional render complete!')
    }, []);

    // arrow function for event handling (could also have been done inline in the jsx if we wanted like below...)
    // onChange={e => setInputText(e.target.value)}
    const handleEvent = event => {
        setInputText(event.target.value);
    }

    return(
        <div>
            <label htmlFor="functionalComponentInput">type to change the state: </label>
            <input id="functionalComponentInput" type="text" onChange={handleEvent}/>
            <p><b>current state:</b> {inputText}</p>
        </div>
    );
}