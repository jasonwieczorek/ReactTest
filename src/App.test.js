import React from 'react';
import renderer from 'react-test-renderer'
import ExampleFunctionalComponent from "./components/ExampleFunctionalComponent";

// test suite
describe('ExampleFunctionalComponent suite', () => {

  it('label renders correctly', () => {

    // renderer just returns a simple javascript object representation of the REACT component
    const component = renderer.create(<ExampleFunctionalComponent inputText='test'/>);
    expect(component.root.findByType('label').props.children).toEqual('type to change the state: ');
  })

  it('props render in input correctly', () => {
    const component = renderer.create(<ExampleFunctionalComponent inputText={'test'}/>);
    expect(component.root.findByType('input').props.value).toEqual('test');
  })

  it('Matches the last test snapshot', () => {
    const component = renderer.create(<ExampleFunctionalComponent inputText={'test'}/>);
    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  })

})

describe('ExampleFunctionalComponent suite', () => {

  // A simple test case, 'it()' is an alias for the test function, they are equivelant
  it('label renders correctly', () => {

    // renderer just returns a simple javascript object representation of the REACT component
    const component = renderer.create(<ExampleFunctionalComponent inputText='test'/>);
    expect(component.root.findByType('label').props.children).toEqual('type to change the state: ');
  })

  //
  it('label renders correctly', () => {
    const component = renderer.create(<ExampleFunctionalComponent inputText={'test'}/>);
    expect(component.root.findByType('input').props.value).toEqual('test');
  })
})
