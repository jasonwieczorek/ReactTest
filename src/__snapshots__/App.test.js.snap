// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleFunctionalComponent suite Matches the last test snapshot 1`] = `
<div>
  <label
    htmlFor="functionalComponentInput"
  >
    type to change the state: 
  </label>
  <input
    id="functionalComponentInput"
    onChange={[Function]}
    type="text"
    value="test"
  />
  <p>
    <b>
      current state:
    </b>
     
    test
  </p>
</div>
`;
